nixos/modules/programs/xss-lock.nix

```diff
@@ -31,6 +31,13 @@ in
       description = "XSS Lock Daemon";
       wantedBy = [ "graphical-session.target" ];
       partOf = [ "graphical-session.target" ];
+      serviceConfig.ExecStartPre = lib.mkIf (config.services.xserver.displayManager.startx.enable) pkgs.writers.writeBash "xss-lock-prereqs" ''
```

Comment:
You're missing parenthesis around writeBash. Anyway, it's better to do:
```suggestion
      preStart = lib.mkIf config.services.xserver.displayManager.startx.enable)  ''
```

Reply:
Thanks, I didn't know about preStart. It runs it as bash? I'll give it a try.

Reply:
Ok, I see. preStart is more or less the same as what I had, except more concise :)